x-pack/test/api_integration/apis/ml/trained_models/get_models.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common_api';
+
+export default ({ getService }: FtrProviderContext) => {
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  describe('GET trained_models', () => {
+    let testModelIds: string[] = [];
+
+    before(async () => {
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      testModelIds = await ml.api.createdTestTrainedModels('regression', 5, true);
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+      // delete created ingest pipelines
+      await Promise.all(testModelIds.map((modelId) => ml.api.deleteIngestPipeline(modelId)));
+    });
+
+    it('returns all trained models with associated pipelined', async () => {
+      const { body } = await supertest
+        .get(`/api/ml/trained_models?with_pipelines=true`)
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(200);
+      // Created models + system model
+      expect(body.length).to.eql(6);
+
+      const sampleModel = body.find((v: any) => v.model_id === 'dfa_regression_model_n_0');
+      expect(Object.keys(sampleModel.pipelines).length).to.eql(1);
+    });
+
+    it('returns models without pipeline in case user does not have required permission', async () => {
+      const { body } = await supertest
+        .get(`/api/ml/trained_models?with_pipelines=true`)
+        .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(200);
+      // Created models + system model
+      expect(body.length).to.eql(6);
+      const sampleModel = body.find((v: any) => v.model_id === 'dfa_regression_model_n_0');
+      expect(sampleModel.pipelines).to.eql(undefined);
+    });
+
+    it('returns trained model by id', async () => {
```

Comment:
Is it worth adding a test for requesting a model for an `id` that doesn't exist?

Reply:
Just added it to the `delete` endpoint tests